### ISSUE TITLE
[CI] ReindexFailureTests testResponseOnSearchFailure failing

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#72033]
This test suite has failed a couple of times now. These failures also seem to cause suite timeouts.

**Build scan:**
https://gradle-enterprise.elastic.co/s/znsrpuwmuj2qs/tests/:modules:reindex:test/org.elasticsearch.index.reindex.ReindexFailureTests/testResponseOnSearchFailure

**Reproduction line:**
`./gradlew ':modules:reindex:test' --tests "org.elasticsearch.index.reindex.ReindexFailureTests.testResponseOnSearchFailure" -Dtests.seed=C566C0B6DC2FF41F -Dtests.locale=ar -Dtests.timezone=Asia/Macau -Druntime.java=8`

**Applicable branches:**
master, 7.x

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.index.reindex.ReindexFailureTests&tests.test=testResponseOnSearchFailure

**Failure excerpt:**
```
java.lang.AssertionError: All incoming requests on node [node_s0] should have finished. Expected 0 but got 524; pending tasks [[{
  "node" : "node_s0",
  "id" : 685,
  "type" : "transport",
  "action" : "internal:index/shard/recovery/start_recovery",
  "description" : "",
  "start_time" : "2021-04-21T12:54:43.996Z",
  "start_time_in_millis" : 1619009683996,
  "running_time" : "1.1m",
  "running_time_in_nanos" : 66355521945,
  "cancellable" : false,
  "headers" : { }
}]]

  at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$43(InternalTestCluster.java:2597)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1019)
  at org.elasticsearch.test.InternalTestCluster.assertRequestsFinished(InternalTestCluster.java:2590)
  at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2565)
  at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:596)
  at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2255)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

**Comment by mark-vieira** [comment]
Here's a similar looking failure. Related?

https://gradle-enterprise.elastic.co/s/egbkulkd5nzq6/tests/:server:internalClusterTest/org.elasticsearch.client.documentation.IndicesDocumentationIT/testPutMappingDocumentation#1

**Comment by mark-vieira** [comment]
And another: https://gradle-enterprise.elastic.co/s/22qxfhl6jv7xo/tests/:server:internalClusterTest/org.elasticsearch.search.suggest.CompletionSuggestSearchIT/testRegex#1

Whatever this root issue is it seems pretty pervasive.

**Comment by jbaiera** [comment]
Looks like we had a similar set of errors error in another test today:

https://gradle-enterprise.elastic.co/s/wzklhwpywjgss

Additionally, we've seen a number of yaml tests fall victim to time outs today across a number of cases. Not sure if that might be generally related to the issues detailed in https://github.com/elastic/elasticsearch/issues/72031

**Comment by jbaiera** [comment]
Aside from the `IndicesOptionsIntegrationIT` reported above, seeing this again in `SystemDataStreamIT` https://gradle-enterprise.elastic.co/s/5og3rxvynyzw2

**Comment by jbaiera** [comment]
And another one on `DestructiveOperationsIT` https://gradle-enterprise.elastic.co/s/ymk3vvzm4ka7u

**Comment by BigPandaToo** [comment]
Looks like this is the same root cause:
https://gradle-enterprise.elastic.co/s/zjoselj5t6nyo/console-log?task=:x-pack:plugin:data-streams:internalClusterTest

**Comment by jaymode** [comment]
This one should be good to close now that fixes have landed in #72153 and #72087 and my search shows no failures due to the same issue in 4 days.
